src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1832,12 +1869,29 @@ private String getCanonicalName0() {
         }
     }
 
+    /**
+     * {@return {@code true} if and only if the underlying class
+     * is an unnamed class}
+     *
+     * @apiNote
+     * An unnamed class is not an {@linkplain #isAnonymousClass anonymous class}.
+     *
+     * @since 21
+     */
+    @PreviewFeature(feature=PreviewFeature.Feature.UNNAMED_CLASSES,
+                    reflective=true)
+    public boolean isUnnamedClass() {
+        return PreviewFeatures.isEnabled() && isSynthetic() && isTopLevelClass();
```

Comment:
Please add a check for final-ness as @mlchung suggested in the other PR of the reflection changes stand-alone.

Reply:
Changing